pydantic/_internal/_generate_schema.py

```diff
@@ -1237,8 +1240,10 @@ def _common_field(
     serialization_alias: str | None = None,
     serialization_exclude: bool | None = None,
     frozen: bool | None = None,
+    field_info: FieldInfo,
     metadata: Any = None,
 ) -> _CommonField:
+    metadata[METADATA_FIELD_INFO_KEY] = field_info
```

Comment:
this seems dangerous since based on this annotation metadata may not be a dict, just pointing it out

Reply:
(You may want to use `pydantic._internal._core_metadata.CoreMetadataHandler` and/or `pydantic._internal._core_metadata.build_metadata_dict`)